### PR TITLE
Perform null check before assigning functionCallContext name.

### LINKED ISSUE #689: Error while typing out a command in between two legal commands
Repro: 
1. Have these commands typed out : 
a) `db.heroes.find({"address.City": "Goooooothammmmmmmmm"
});`
b)`db.heroes.drop();`

2. Start typing a command in between these commands. 
a)`db` -> Fine
b)`db.heroes` -> Fine
c)`db.herores.` -> Error before intellisense: "TypeError: Cannot read property 'text' of undefined"

Error stems from visitFunctionCall that tries to look up `ctx._FUNCTION_NAME.text` of the string "" (the empty string after `db.heroes`).

### PR DESCRIPTION
**Pull request by PrashanthCorp**
Fixes #689

### CODE REVIEW

File: src/mongo/MongoScrapbook.ts [hunk 1]
```diff
@@ -201,7 +201,7 @@ class FindMongoCommandsVisitor extends MongoVisitor<MongoCommand[]> {
 	}
 
 	visitFunctionCall(ctx: mongoParser.FunctionCallContext): MongoCommand[] {
-		if (ctx.parent instanceof mongoParser.CommandContext) {
+		if (ctx.parent instanceof mongoParser.CommandContext && ctx._FUNCTION_NAME) {
 			this.commands[this.commands.length - 1].name = (ctx._FUNCTION_NAME && ctx._FUNCTION_NAME.text) || "";
```
Comment (nturinski) [inline src/mongo/MongoScrapbook.ts]: nit: I don't think you need to check ctx._FUNCTION_NAME on line 205 if you're doing it on 204.

Comment (nturinski) [inline src/mongo/MongoScrapbook.ts]: What does `(ctx._FUNCTION_NAME && ctx._FUNCTION_NAME.text)` resolve to anyway?  Are those 2 strings?

File: src/mongo/MongoScrapbook.ts [hunk 2]
```diff
@@ -201,7 +201,7 @@ class FindMongoCommandsVisitor extends MongoVisitor<MongoCommand[]> {
 	}
 
 	visitFunctionCall(ctx: mongoParser.FunctionCallContext): MongoCommand[] {
-		if (ctx.parent instanceof mongoParser.CommandContext) {
+		if (ctx.parent instanceof mongoParser.CommandContext && ctx._FUNCTION_NAME) {
```
Comment (StephenWeatherford) [inline src/mongo/MongoScrapbook.ts]: Not sure we should be avoiding creating a command here.  Need to play around with it.

File: src/mongo/MongoScrapbook.ts (hunk 1 above)
Comment (PrashanthCorp) [reply src/mongo/MongoScrapbook.ts]: `ctx._FUNCTION_NAME` is a Token that the parser identifies. 
`ctx._FUNCTION_NAME.text` is the string the token points to. 

Should resolve to a string, else a false-y value.

**Comment by PrashanthCorp** [comment]
I'm closing this. 

It looks like #660  fixes the issue anyway. 
The corresponding test was not present in the PR.

**Pull request closed without merging**